app/controllers/api/base_controller.rb

```
@@ -48,17 +48,48 @@ def resource_class
       @resource_class ||= resource_name.classify.constantize
     end
 
-    def resource_scope(controller = controller_name)
+    def resource_scope(controller = controller_name, options = {})
       @resource_scope ||= begin
-        scope = resource_class.scoped
-        if resource_class.respond_to?(:authorized)
-          scope.authorized("#{action_permission}_#{controller}", resource_class)
+        if nested_obj
+          resource_scope_nested(controller, options)
         else
-          scope
+          resource_scope_not_nested(controller, options)
         end
       end
     end
 
+    def resource_scope_not_nested(controller = controller_name, options = {})
+      scope = resource_class.scoped
+      options[:permission] ||= "#{action_permission}_#{controller}"
+      if resource_class.respond_to?(:authorized)
+        scope.authorized(options[:permission], resource_class)
+      else
+        scope
+      end
+    end
+
+    def resource_scope_nested(controller = controller_name, options = {})
+      raise KeyError unless (options.keys - [:association, :permission, :includes]).empty?
+
+      options[:association] ||= controller
+      options[:permission]  ||= "#{action_permission}_#{options[:association]}"
+      if nested_obj
+        if nested_obj.respond_to?(options[:association]) && nested_obj.send(options[:association]).respond_to?(:authorized)
+          nested_obj.send(options[:association]).authorized(options[:permission], resource_class)
+        elsif nested_obj.respond_to?(options[:association])
+          nested_obj.send(options[:association])
+        else
+          resource_scope_not_nested(controller, options)
```

same here, this branch of `ifs` can silently ignore :association option, it would be better to restructure it here in a way that options is authoritative in what methods resource has to support.